lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
Mh, I didn't know that `quote do: msg` and `quote do: msg()` produce different ASTs. Then yes, we need to ensure we don't call any 0-arity functions. Any suggestions?